### PR TITLE
add functions and pytest

### PR DESCRIPTION
**Pull request by lyiris22**
- Add contents for function `my_gather` and `my_spread`
- Fix some test cases for `test_my_gather` and `test_my_spread`
- Add `setup.py`
- Add code, modify folder name and structure to run pytest

### CODE REVIEW

File: miniTidyPy/my_gather.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# This is the script for function my_spread
```
Comment (ellognea) [inline miniTidyPy/my_gather.py]: I think you meant "This is the script for my function "my_gather"

File: miniTidyPy/my_gather.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+# Function left black at this point for further contributions
```
Comment (ellognea) [inline miniTidyPy/my_gather.py]: Can this be removed now that the function is defined ?

**Review by ellognea — CHANGES_REQUESTED** [review]
Great job Iris !  my_gather function has been neatly written. 
Just a few comment changes -- see comments.

File: miniTidyPy/my_spread.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    Returns:
+        new_df (dataframe): the transformed new dataframe
+    '''
+
+    # check if input is a dataframe
+    if not isinstance(df, pd.DataFrame):
+        return None
+
+    # check if key and value are in list(df)
+    if key not in list(df):
```
Comment (huijuechen) [inline miniTidyPy/my_spread.py:29]: Should we add message here? Something like "original df kept, did not get spread" (def more formal)? Sometimes df can contain many columns, might be difficult to check through all.

Comment (huijuechen) [reply miniTidyPy/my_spread.py:29]: Overall structure looks clean!

**Pull request merged**